Client/Clique/Modules/EventDetail/EventDetailViewController.swift

```diff
@@ -45,7 +53,77 @@ private extension EventDetailViewController {
         } else {
             _eventCost.text = Constants.Labels.labelFree
         }
-        
         _eventCategory.text = _event.category
+        _buttonJoinEvent.addTarget(self, action: #selector(_registerOnEvent), for: .touchUpInside)
+        if(_stringToTimeStamp(timeString: _event.timestamp) < NSDate().timeIntervalSince1970){
+            _didItEnd = true
+        }
+        _checkUserStatusOnEvent()
+    }
+    
+    func _stringToTimeStamp(timeString: String) -> TimeInterval {
+        let dateFormatterGet = DateFormatter()
+        dateFormatterGet.dateFormat = "dd/MM/yyyy HH:mm:ss"
+        let date = dateFormatterGet.date(from: timeString)
+        print(date!.timeIntervalSince1970)
+        return date!.timeIntervalSince1970
+    }
+    
+    func _checkUserStatusOnEvent(){
+        _eventServices.checkUserStatusOnEvent(event_id: _event_Id){ [weak self] result in
+            guard let _self = self else { return }
+            switch result{
+            case .success(let status):
+                _self._status = status
+                _self._activateButton()
+            case .failure(let error):
+                print(error)
+            }
+        }
+    }
+    
+    func _activateButton(){
+        if(!_didItEnd){
+            if(_status == 1 || _status == 3){
+                        _buttonJoinEvent.setTitle("Join", for: .normal)
+                        _buttonJoinEvent.isHidden = false
+                        _alertMessage = "Are you sure you want to join this event?"
+                    } else if(_status == 2) {
+                        _buttonJoinEvent.setTitle("Cancel", for: .normal)
+                        _buttonJoinEvent.isHidden = false
+                        _alertMessage = "Are you sure you want to cancel this event?"
+                    }
+        } else if(_didItEnd && _status == 2) {
+            print("Ovdje bi trebalo ici ocjenjivanje")
+        } else {
+            _buttonJoinEvent.isHidden = true
+        }
+    }
+    
+    @objc func _registerOnEvent(){
+        let refreshAlert = UIAlertController(title: "Confirmation", message: _alertMessage, preferredStyle: UIAlertController.Style.alert)
+
+        refreshAlert.addAction(UIAlertAction(title: "Ok", style: .default, handler: { (action: UIAlertAction!) in self.reg()
+        }))
+
+        refreshAlert.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: { (action: UIAlertAction!) in
+              print("Handle Cancel Logic here")
+        }))
+
+        present(refreshAlert, animated: true, completion: nil)
```

Comment:
I've seen that localizations are defined in the `Constants.swift` file, so do the same accordingly.